src/compiler/core.ts

```diff
@@ -1122,17 +1122,6 @@ export function* arrayReverseIterator<T>(array: readonly T[]) {
     }
 }
 
-/**
- * Stable sort of an array. Elements equal to each other maintain their relative position in the array.
- *
- * @internal
- */
-export function stableSort<T>(array: readonly T[], comparer: Comparer<T>): SortedReadonlyArray<T> {

```

Comment:
I should note that there's a behavior change with this PR; sort is:

```
export function sort<T>(array: readonly T[], comparer?: Comparer<T>): SortedReadonlyArray<T> {
    return (array.length === 0 ? array : array.slice().sort(comparer)) as SortedReadonlyArray<T>;
}
```

So... if you happen to sort an empty array, you don't make a copy! This is scary but nothing fails either way and I don't know how to feel about that. This feels like a footgun that kinda makes me scared of this PR, but no more scared than I already am about using `sort` at all.

Reply:
Does anything break if you replace it with

```ts
export function sort<T>(array: readonly T[], comparer?: Comparer<T>): SortedReadonlyArray<T> {
    return (array.length === 0 ? Object.freeze([]) : array.slice().sort(comparer)) as SortedReadonlyArray<T>;
}
```

Reply:
I'll check, though that does make me want to freeze `emptyArray` and then return that instead.

Reply:
At that point though, there's not really a reason not to slice the array if we know it's empty...

Reply:
Oops, sorry, no, your code fails lots of tests. Your code freezes the input array so breaks callers who are just sorting lists they have.

Reply:
So now that I realize that the return result is readonly, I'm no longer concerned about the safety of this PR. I will send another PR which tries out freezing things, though, as if its perf is fine that's nice to prevent oopses.